include/camp/resource.hpp

```diff
@@ -93,35 +93,71 @@ namespace resources
       void wait_for(Event *e) { m_value->wait_for(e); }
       void wait() { m_value->wait(); }
 
+      size_t get_id() const {
```

Comment:
This ought to be either private or protected IMO. We can have the functions for hashing as friends, but exposing the id could easily lead to unintended coupling in user code.